tests/integration/kubernetes/confidential_common.sh

```diff
@@ -44,18 +44,43 @@ function check_hypervisor_for_confidential_tests() {
 	local kata_hypervisor="${1}"
 	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
 	# having substrings, like qemu, being matched with qemu-$something.
-    if [[ " ${SUPPORTED_TEE_HYPERVISORS[*]} " =~ " ${kata_hypervisor} " ]] ||\
+    if check_hypervisor_for_confidential_tests_tee_only "${kata_hypervisor}" ||\
        [[ " ${SUPPORTED_NON_TEE_HYPERVISORS[*]} " =~ " ${kata_hypervisor} " ]]; then        
         return 0
     else
         return 1
     fi
 }
 
-# Common setup for confidential tests.
-function confidential_setup() {
+# This function verifies whether the input hypervisor supports confidential tests and 
+# relies on `KATA_HYPERVISOR` being an environment variable
+function check_hypervisor_for_confidential_tests_tee_only() {
+	local kata_hypervisor="${1}"
+	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
+	# having substrings, like qemu, being matched with qemu-$something.
+	if [[ " ${SUPPORTED_TEE_HYPERVISORS[*]} " =~ " ${kata_hypervisor} " ]]; then
+		return 0
+	else
+		return 1
+	fi
+}
+
+# Common check for confidential tests.
+function is_confidential_runtime_class() {
 	ensure_yq
```

Comment:
I appreciate this wasn't your change, but why do we need yq for `is_confidential_runtime_class` and `is_confidential_hardware`?

Reply:
IIRC originally `confidential_setup()` was conceived as a common setup function (not only to skip tests). So in commit 2c0bc8855b we added this line to ensure_yq. The commit doesn't say why and I found yq being only used for yaml instrumentation for policy tests. Besides, `tests/integration/kubernetes/run_kubernetes_tests.sh` also call `ensure_yq` so I suspect we are safe to remove that call from here.